scripts/run_tempest.yml

```diff
@@ -13,23 +13,41 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+# Get defcore tests and set tempest_test_whitelist fact
+- name: Get defcore tests and set tempest_test_whitelist fact
+  hosts: utility_all[0]
+  gather_facts: no
+  user: root
+  tasks:
+    - include: get_defcore_tests.yml
+    - name: Set tempest_test_whitelist fact
+      set_fact:
+        tempest_test_whitelist: "{{ tempest_test_whitelist + defcore_tests }}"
+  tags:
+    - tempest_install
+  vars:
+    # 2017.01 corresponds to Ocata
+    version_number: "2017.01"
+    defcore_url: "https://raw.githubusercontent.com/openstack/defcore/master/{{ version_number }}.json"
 
 # Run the tempest playbook to install tempest
 - include: ../openstack-ansible/playbooks/os-tempest-install.yml
   tags:
     - tempest_install
+  vars:
+    tempest_run: no
 
-- name: Execute Tempest Tests
-  hosts: utility[0]
+# Run the tempest tests from the os_tempest role
+- name: Run tempest tests from the os_tempest role
+  hosts: utility_all[0]
+  gather_facts: no
   user: root
-  tasks:
-    - name: Execute tempest tests
-      shell: |
-        export RUN_TEMPEST_OPTS={{ tempest_run_tempest_opts | join(' ') }}
-        export TESTR_OPTS={{ tempest_testr_opts | join(' ') }}
-        bash /opt/openstack_tempest_gate.sh {{ tempest_test_sets }}
-      changed_when: false
   tags:
     - tempest_execute_tests
-
-
+  tasks:
+    - name: Run Tempest Tests
+      include_role:
+        name: os_tempest
+        tasks_from: tempest_run
+  vars:
```

Comment:
This is unnecessary as you're running the task file directly, unless I'm missing some reason for the conditional inclusion?